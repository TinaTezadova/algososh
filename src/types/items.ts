import { ElementStates } from './element-states';

export type ArrayItem<T = string> = {
  id: string;
  value: T;
  state: ElementStates;
  head?: string | null;
  tail?: string | null;
  gone?: boolean
};

export type SetState = React.Dispatch<React.SetStateAction<ArrayItem[]>>;

export interface ListNode<T> {
  id: string;
  state: ElementStates;
  next: ListNode<T> | null;
  value: T;
};