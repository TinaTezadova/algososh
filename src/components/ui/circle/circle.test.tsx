import { Circle } from './circle';
import { render, screen } from '@testing-library/react';
import { ElementStates } from '../../../types/element-states';

describe('Тестирвание компонента Circle', () => {
    it('Рендер компонента без текста', () => {
        render(<Circle />);
        const circle = screen.getByTestId('circle-wrapper');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента с текстом', () => {
        render(<Circle letter="Text"/>);
        const circle = screen.getByTestId('circle-wrapper');
        expect(circle).toHaveTextContent('Text');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента с текстом в head', () => {
        render(<Circle head="Text"/>);
        const circle = screen.getByTestId('circle-wrapper');
        expect(circle).toHaveTextContent('Text');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента с другим реакт-компонентом в head', () => {
        render(<Circle head={<p>Text</p>}/>);
        const circle = screen.getByTestId('circle-wrapper');
        expect(circle).toHaveTextContent('Text');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента с текстом в tail', () => {
        render(<Circle tail="Text"/>);
        const circle = screen.getByTestId('circle-wrapper');
        expect(circle).toHaveTextContent('Text');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента с другим реакт-компонентом в tail', () => {
        render(<Circle tail={<p>Text</p>}/>);
        const circle = screen.getByTestId('circle-wrapper');
        expect(circle).toHaveTextContent('Text');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента с index', () => {
        render(<Circle index={3}/>);
        const circle = screen.getByTestId('circle-wrapper');
        expect(circle).toHaveTextContent('3');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента с пропсом isSmall', () => {
        render(<Circle isSmall/>);
        const circle = screen.getByTestId('circle');
        expect(circle).toHaveClass('small');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента в состоянии default', () => {
        render(<Circle state={ElementStates.Default} letter="defaultState"/>);
        const circle = screen.getByTestId('circle');
        expect(circle).toHaveTextContent('defaultState');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента в состоянии changing', () => {
        render(<Circle state={ElementStates.Changing} letter="changingState"/>);
        const circle = screen.getByTestId('circle');
        expect(circle).toHaveTextContent('changingState');
        expect(circle).toMatchSnapshot();
    });

    it('Рендер компонента в состоянии modified', () => {
        render(<Circle state={ElementStates.Modified} letter="modifiedState"/>);
        const circle = screen.getByTestId('circle');
        expect(circle).toHaveTextContent('modifiedState');
        expect(circle).toMatchSnapshot();
    });

});