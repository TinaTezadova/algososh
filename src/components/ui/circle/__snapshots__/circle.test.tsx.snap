// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирвание компонента Circle Рендер компонента без текста 1`] = `
<div
  class="content "
  data-testid="circle-wrapper"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента в состоянии changing 1`] = `
<div
  class="circle   changing"
  data-testid="circle"
>
  <p
    class="text text_type_circle text_color_input letter"
  >
    changingState
  </p>
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента в состоянии default 1`] = `
<div
  class="circle   default"
  data-testid="circle"
>
  <p
    class="text text_type_circle text_color_input letter"
  >
    defaultState
  </p>
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента в состоянии modified 1`] = `
<div
  class="circle   modified"
  data-testid="circle"
>
  <p
    class="text text_type_circle text_color_input letter"
  >
    modifiedState
  </p>
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента с index 1`] = `
<div
  class="content "
  data-testid="circle-wrapper"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    3
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента с другим реакт-компонентом в head 1`] = `
<div
  class="content "
  data-testid="circle-wrapper"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <p>
      Text
    </p>
  </div>
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента с другим реакт-компонентом в tail 1`] = `
<div
  class="content "
  data-testid="circle-wrapper"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <p>
      Text
    </p>
  </div>
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента с пропсом isSmall 1`] = `
<div
  class="circle  small default"
  data-testid="circle"
>
  <p
    class="text text_type_circle text_color_input letter"
  />
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента с текстом 1`] = `
<div
  class="content "
  data-testid="circle-wrapper"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      Text
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента с текстом в head 1`] = `
<div
  class="content "
  data-testid="circle-wrapper"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    Text
  </div>
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирвание компонента Circle Рендер компонента с текстом в tail 1`] = `
<div
  class="content "
  data-testid="circle-wrapper"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Text
  </div>
</div>
`;
